examples/DataflowExamples/HelloGraph/main.cpp

```diff
@@ -0,0 +1,146 @@
+#include <Dataflow/Core/DataflowGraph.hpp>
+#include <Dataflow/Core/Nodes/Functionals/FilterNode.hpp>
+#include <Dataflow/Core/Nodes/Sinks/SinkNode.hpp>
+#include <Dataflow/Core/Nodes/Sources/CoreDataSources.hpp>
+
+using namespace Ra::Dataflow::Core;
+
+/* ----------------------------------------------------------------------------------- */
+/**
+ * \brief Demonstrate how to use a graph to filter a collection
+ */
+int main( int argc, char* argv[] ) {
+    using RaVector = Ra::Core::VectorArray<Scalar>;
+    //! [Creating an empty graph]
+    DataflowGraph g { "helloGraph" };
+    //! [Creating an empty graph]
+
+    //! [Creating Nodes]
+    auto sourceNode    = new Sources::SingleDataSourceNode<RaVector>( "Source" );
```

Comment:
Nodes are own by the graph. 
The documentation should be clearer about this. 
There is an ownership transfer from the caller to the graph when nodes are added to a graph, and only if they are added. 
So, if we move a unique_ptr into the graph but the node could'nt be added, what happens on the unique_ptr ? 
We can add a function to test if a node can be added to a graph and then use this result as requirements of the addnode method. In this case, we can move a unique_ptr and return a optional with the naked pointer to the node.

Reply:
Node can't be added only if thereis a name conflict as node names should be unique in a graph in order to set links. So, in this exemple, everything goes fine. If a node could'nt be added, the user can test the return of add_node, change the name and add the node again. The raw pointer given to the graph is used to create a unique_ptr  managed by the graph(ownership is transfered to the graph) but the user can still use the pointer to further add links, set state, etc ... 
To prevent some problems, resulting from the current implementation or from the deletion of a unique pointer not added to the graph, I'll change the add_node that will return a pair <bool, raw ptr>. If the bool is true, the node was added to the graph and the user can use the raw pointer without ownership management (the graph owns it). If the bool is false, the raw pointer will be own by the caller and it can fix the name, embed the pointer in a unique ptr and add it again.
We can also change the ways links are defined to use the uuid (which is unique by construction) of the node instead of its name, then allowing non unique names and but it is clearly less user friendly ^^ and do not prevent all collisions if aliases are still allowed. So yep, I'll go for unique_ptr transfer with a returned pair